Improve setup script output and functional behavior

NxFerrara opened the pull request:
## Changes:
- Updated status message format
- Changed behavior of scripts to exit on any errors
- Suppress command standard output

## Considerations:
- Need to test whether `set -e` since it is a powerful flag (exits scripts on all errors).

NxFerrara committed:
Fixing setup script status message format

Remove "\t" from all script status messages to make the spacing consistent.

NxFerrara committed:
Remove unnecessary print usage

Printing the usage of the setup script is not relevant for when the environment variables are not sourced correctly.

NxFerrara committed:
Add flag in setup scripts to exit on all errors

`set -e` makes it so any non-zero exit status on any command will make the script exit with the same exit code.

NxFerrara committed:
Remove redundant source

Since the setup.sh script that calls all of these setup scripts sources the scripts_config.sh, it becomes redundant to source it again.

NxFerrara committed:
Squashed commit of the following:

commit d078b20fc4373fd497337076a3e8017cd5e254df
Author: Juan Becerra <jdb9056@nyu.edu>
Date:   Mon Mar 4 21:31:15 2024 -0500

    Fix warning and inconsistencies in 5_set_device_id

    Mostly adding double-quotes to variable substitution statements to avoid
    globbing and word splitting. Separated declaration and assignment of
    variables in the same line to avoid masking exit status.

    Note: When you declare and assign a variable in the same command using
          command substitution, the exit status of the command gets
          overwritten. Only the exit status of the assignment (which is
          typically successful, returning 0) is available, not the exit
          status of the command inside the command substitution.

commit 4641dc57c58b310dd45740d82fd28a5ffe6ecd25
Author: Juan Becerra <jdb9056@nyu.edu>
Date:   Mon Mar 4 19:01:28 2024 -0500

    Fix reference to DATABASE_URL

commit a9074e3a8dcbbd37e4f4f67bb54bc6916de8a9ad
Merge: 01af904 6b3f020
Author: Juan Becerra <jdb9056@nyu.edu>
Date:   Mon Mar 4 18:56:21 2024 -0500

    Remove unnecessary comments

commit 01af9047e8e124e03e0cd8ab613dc1b9de398175
Author: Juan Becerra <jdb9056@nyu.edu>
Date:   Mon Mar 4 18:31:31 2024 -0500

    Refactor get_new_device_id to fetch first unallocated device ID

    Logic in get_new_device_id() was changed to fetch the first available
    (unallocated) device ID from the database instead of generating a new
    ID based on the last record's ID. This approach allows for repurposing
    device IDs that were once taken but have been freed by a device.

    In order to enable this behavior, an 'allocated' column was created in
    the 'devices' table. This column serves to identify whether a particular
    device ID was allocated. With this in mind, insert_device_id() was
    modified to include an 'allocated' field in the database insert query,
    marking the device as allocated upon insertion.

    As of this commit, this implementation does not handle race conditions.

NxFerrara committed:
Suppress the standard output from all commands

The only output that is relevant are the error outputs since there are already dedicated messages to describe the script progress.

NxFerrara committed:
Set e flag dynamically in setup.sh

Since we don't want the setup.sh script to exit upon a child script exiting with a non-zero status as we want it to print "failed" first, it makes sense to change the flag during child script execution.

NxFerrara committed:
Merge branch 'jbesoto/setup-review' into nxferrara/setup-review

jbesoto commented on scripts/setup/post_reboot/1_create_pipes.sh:
```diff
@@ -1,14 +1,17 @@
 #!/bin/bash
 
+# This tells the shell to exit the script if any command within the script exits with a non-zero status
```
Unnecessary comment, purpose of the flag is self-evident. Please remove all cases.

kc34251 replied:
I would agree that it is irrelevant to put in each of the files but this flag's purpose could be mentioned as a comment in setup.sh

NxFerrara replied:
I think that it is easily searchable as it is a built-in flag, so I agree with @jbesoto. I will omit all comments.

jbesoto commented on scripts/setup/post_reboot/1_create_pipes.sh:
```diff
@@ -1,14 +1,17 @@
     # Root check
     if [ "$(id -u)" -ne 0 ]; then
-        echo -e "\t${WARNING_MSG} This script must be run as root. Please use sudo."
+        echo -e "${WARNING_MSG} This script must be run as root. Please use sudo."
```
No escape or special characters are used, so `-e` is not needed. Simple fix, address all other cases.

NxFerrara replied:
Does this apply even with the variables like `WARNING_MSG` that have special characters?

NxFerrara replied:
Yes it does apply. The commands used for formatting already deal with echo processing correctly.

jbesoto commented on scripts/setup/post_reboot/1_create_pipes.sh:
```diff
@@ -17,17 +20,17 @@ create_fifo_pipes() {
     else
-        mkfifo "${TOUCH_SENSOR_TO_SCREEN_PIPE}"
-        echo -e "\t\t${OK_MSG} FIFO pipe '${TOUCH_SENSOR_TO_SCREEN_PIPE}' created."
+        mkfifo "${TOUCH_SENSOR_TO_SCREEN_PIPE}" > /dev/null
```
Not sure why this was done since `mkfifo` does not send any output to `stdout`.

NxFerrara replied:
Totally agree, I removed all such cases.

jbesoto commented on scripts/setup/post_reboot/3_connect_wifi.sh:
```diff
@@ -11,7 +14,7 @@ connect_wifi() {
     local max_attempts=5
```
This is unrelated to this PR, but this should be set to `readonly`, like so:
```diff
local max_attempts=5
+ readonly max_attempts
```

NxFerrara replied:
Makes sense, I have made the change.

jbesoto commented on scripts/setup/pre_reboot/1_set_device_overlays.sh:
```diff
@@ -1,20 +1,23 @@
 
 set_custom_dts() {
-    cp "${PROJECT_PATH}/custom/spi-cc-1cs-ili9341.dts" "${PROJECT_DIR}/libretech-wiring-tool/libre-computer/aml-s905x-cc/dt/spi-cc-1cs-ili9341.dts" || { echo "Install script failed"; exit 1; }
+    cp "${PROJECT_PATH}/custom/spi-cc-1cs-ili9341.dts" "${PROJECT_DIR}/libretech-wiring-tool/libre-computer/aml-s905x-cc/dt/spi-cc-1cs-ili9341.dts" > /dev/null
```
`cp` does not send any output to `stdout`. Redirection is unnecessary

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/pre_reboot/1_set_device_overlays.sh:
```diff
@@ -26,8 +29,8 @@ reset_overlays() {
             [Yy] ) echo "Resetting overlays...";
-                rm -f "${OVERLAY_MERGED_FLAG}"
-                /opt/libretech-wiring-tool/ldto reset
+                rm -f "${OVERLAY_MERGED_FLAG}" > /dev/null
```
Redirection is unnecessary for `rm`

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/pre_reboot/6_setup_logging.sh:
```diff
@@ -1,9 +1,11 @@
 
-touch ${LOGROTATE_PATH}/${LOGCONF_FILENAME}
+# Setup logrotate config file
+touch ${LOGROTATE_PATH}/${LOGCONF_FILENAME} > /dev/null
```
Redirection is unnecessary for `touch`. No output

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/pre_reboot/6_setup_logging.sh:
```diff
@@ -1,9 +1,11 @@
 # Empty log file is required by the logrotate tool
-touch ${PROJECT_PATH}/.app/logs/${LOG_FILENAME}
\ No newline at end of file
+touch ${PROJECT_PATH}/.app/logs/${LOG_FILENAME} > /dev/null
```
Same here

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/setup.sh:
```diff
@@ -22,15 +25,17 @@ run_scripts() {
     for script in "${target_dir}"/*.sh; do
         # TODO: Add name script name 
-        # TODO: 
+        # TODO: Test whether the script exiting with non-zero status will be caught by set -e (We don't want this to happen)
```
I thought this is the behavior we wanted: non-zero status causes the script to exit.

NxFerrara replied:
Well, we want the error message to print first in the setup.sh script since the behavior is still exiting if the status of the child script is non-zero. Let me know if I am misinterpreting the code/behavior.

jbesoto commented on scripts/setup/setup.sh:
```diff
@@ -52,18 +57,18 @@ run_pre_reboot_tasks() {
                 # Create file flags and locks required during post-reboot setup
-                touch "${DISPLAY_FRAME_BUFFER_LOCK_PATH}"
-                touch "${PRE_REBOOT_FLAG}"
+                touch "${DISPLAY_FRAME_BUFFER_LOCK_PATH}" > /dev/null
```
Redirection is unnecessary for `touch`

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/setup.sh:
```diff
@@ -52,18 +57,18 @@ run_pre_reboot_tasks() {
-                touch "${DISPLAY_FRAME_BUFFER_LOCK_PATH}"
-                touch "${PRE_REBOOT_FLAG}"
+                touch "${DISPLAY_FRAME_BUFFER_LOCK_PATH}" > /dev/null
+                touch "${PRE_REBOOT_FLAG}" > /dev/null
```
Same here

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/setup.sh:
```diff
@@ -52,18 +57,18 @@ run_pre_reboot_tasks() {
 
                 if [ -f "${REBOOT_HALTED_FLAG}" ]; then
-                    rm -f "${REBOOT_HALTED_FLAG}"
+                    rm -f "${REBOOT_HALTED_FLAG}" > /dev/null
```
Redirection is unnecessary for `rm`

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/setup.sh:
```diff
@@ -52,18 +57,18 @@ run_pre_reboot_tasks() {
                 ;;
             [Nn])
-                touch "${REBOOT_HALTED_FLAG}"
+                touch "${REBOOT_HALTED_FLAG}" > /dev/null
```
Redirection is unnecessary for `touch`

NxFerrara replied:
Removed.

jbesoto commented on scripts/setup/setup.sh:
```diff
@@ -88,17 +93,17 @@ run_post_reboot_tasks() {
+    mkdir -p ${PROJECT_PATH}/.app/flags > /dev/null
+    mkdir -p ${PROJECT_PATH}/.app/locks > /dev/null
+    mkdir -p ${PROJECT_PATH}/.app/logs > /dev/null
+    mkdir -p ${PIPE_FOLDER_PATH} > /dev/null
```
Redirection is unnecessary for `mkdir`

NxFerrara replied:
Removed.

jbesoto reviewed (changes requested):
Good work. Really good initiative setting these flags to modify the script's behavior in such a way that it meets our needs. Mostly issues related to unnecessary redirection. Let it rip.
-jdb

kc34251 reviewed (changes requested):
I agree with Juan on removing all the unnecessary redirections for rm, mkdir and touch and also removing the comments on each file explaining the error flag, but I feel that it could be stated somewhere in the setup.sh script

NxFerrara committed:
Remove set -e comment

The use of the flag is self-evident, so no comment needed.

NxFerrara committed:
Remove unnecessary redirection to /dev/null

Some commands do not print to standard output by default, so removing suppression accordingly.

NxFerrara committed:
Make variable read-only

max_attempts should be readonly according to Juan.

NxFerrara committed:
Remove unnecessary -e flag for echo

Remove the -e flag for messages without special characters since it is not needed.

NxFerrara reviewed (commented):
Good.

jbesoto reviewed (approved):
Good job

kc34251 reviewed (approved):
Good Work
